titus_isolate/utils.py

```diff
@@ -193,7 +195,12 @@ def start_periodic_scheduling():
 
 
 def is_kubernetes() -> bool:
```

Comment:
I'll do you one better.
```

from functools import lru_cache

@lru_cache()
def is_kubernetes():
  ...
```

Example:
```
>>> @lru_cache()
... def foo():
...     print('hi!')
...     return 'hi'
... 
>>> 
>>> foo()
hi!
'hi'
>>> foo()
'hi'
```

Reply:
Or use cachetools.

```
import cachetools.func
import time

@cachetools.func.ttl_cache(maxsize=1000, ttl=10)
def foo(val : str) -> str:
  print(f'received call for {val}')
  return val + '-output'

print('Initially fetching value')
assert foo('test') == 'test-output'
print('Fetching value again, should be cached')
assert foo('test') == 'test-output'
print('Sleeping')
time.sleep(11)
print('Fetching value again 11 seconds later, should not be cached')
assert foo('test') == 'test-output'
```

https://repl.it/@SargunDhillon/OnlyEarnestYottabyte

Reply:
magic.  👎 
The LRU cache is also really for functions that take arguments and can return more than 2 [true | false] values.  Further it's a per-function cache.  Probably there are other config values we'll want to cache and we'll probably want one cache, not N caches for storing them.

Reply:
Yeah, I was thinking you'd add ttl_cache to: https://github.com/Netflix-Skunkworks/titus-isolate/blob/2c15ab4b8fc9aff1630a7b88c3f4da15785ac6b2/titus_isolate/config/agent_property_provider.py#L11 -- This seems like a way to avoid this problem happening in the future.

Reply:
I added `lru_cache` to the ConfigManager.